### ISSUE TITLE
Issues with trakt now putting watched movies into database

### ISSUE THREAD

**Issue by ThatSlimGuy** [razzeee/script.trakt#504]
Been having an issue where trakt will simply not put a movie or tv show as watched into trakt unless the file is watched to the very last second of the file. Even with it being set to rate the track after 75% of the file has been watched, the rating will not pop up, and will say as watched in Kodi, but not add as watched in Trakt. I have to force sync the library for them to then sync to trakt.

Is happening on my Android version and also happening on the side loaded Apple Tv version.

Does this mean it will be an issue with my trakt account or is something messed within the addons ?

**Comment by razzeee** [comment]
Which version are you running? Any logfile?

**Comment by ThatSlimGuy** [comment]
> Which version are you running? Any logfile?

Currently running the latest update on AndroidTV being 18.7 for Kodi
Trakt seems to be the latest at 3.2.5

But as i said above the same error is on older Krypton version which is the only option available for AppleTV atm. So it makes me feel my Trakt is the issue ?

**Comment by razzeee** [comment]
Can't really say. You should look at your logfile after having it fail.

**Comment by ThatSlimGuy** [comment]
The issue is so strange. If i skip through a file, and then let the last 1 minute play, i stop it, the prompt comes up, i can rate, but there is a huge delay until the confirmation comes up it has been successful.

But if i let a 20 minute episode play, ill stop at the last 30 seconds, and the rating never ever pops up. But, if i let the file play until the very end of the file, it will pop.

**Comment by ThatSlimGuy** [comment]
[endlog.txt](https://github.com/trakt/script.trakt/files/4842576/endlog.txt)

Does any of this look wrong ?

**Comment by razzeee** [comment]
No, looks normal to me.

**Comment by ThatSlimGuy** [comment]
That file was the one that popped up fine at the end but i skipped half the file. Im currently letting an episode run now and will up that log in about 15mins. Is there any signs in that log to show why the confirmation of rating takes like 20-25 seconds ?

**Comment by razzeee** [comment]
That's probably due to trakt being slow. As far as I can tell. 

https://trakt.docs.apiary.io/reference/scrobble/stop/stop-or-finish-watching-in-a-media-center

**Comment by ThatSlimGuy** [comment]
Ok give me about 10mins or so, i will see what the log does with an episode playing start to finish

**Comment by ThatSlimGuy** [comment]
[enderro.txt](https://github.com/trakt/script.trakt/files/4842616/enderro.txt)
Yep, did it again. Never pops up after episode. Marks episode as watched, never pops the ratings or marks as watched in trakt

**Comment by ThatSlimGuy** [comment]
[trakt.txt](https://github.com/trakt/script.trakt/files/4842673/trakt.txt)
is this something ?

**Comment by razzeee** [comment]
Yeah, but I'm not sure where that originates from.

**Comment by ThatSlimGuy** [comment]
This should help

**Comment by ThatSlimGuy** [comment]
You think its my trakt account messing it up somehow ?

**Comment by ThatSlimGuy** [comment]
Tried so many things and can't seem to get around it. Trying to avoid a full reinstall of everything. You have any ideas on what I could do?

**Comment by razzeee** [comment]
I would either let the trakt guys check your account or maybe open an issue on https://github.com/fuzeman/trakt.py/ as that's where that error originates.

**Comment by ThatSlimGuy** [comment]
Seems there is a lot of issues as well with RAR files again also with the new kodi update recently, i ended up just trying a full reinstall cause it was hard with the limited time i have to work on trying to fix it, and also find a consistent conversation where i could get feedback about my issue. Hoping a new install will fix it all.

**Comment by ThatSlimGuy** [comment]
Could you help explain what the issue you saw was or just a bit more Info about it so I can try and find a solution. It's driving me insane cause there is no real was to talk to others about helping solve the issue

**Comment by razzeee** [comment]
@rectifyer 

His user is getting a "connection reset by peer"

**Comment by rectifyer** [comment]
Hmm, not exactly sure what would cause that. Could be heavy load on the API servers. @ThatSlimGuy are you running a VPN or anything similar that you can try routing around to see if that changes anything? Does your data load ok on the Trakt website?

**Comment by ThatSlimGuy** [comment]
Not running a vpn at all and don't have any issues with website at all. Loads fine on any device I have. Strange thing is I can let a file run until the very very end and 10 times outta 10, it will work fine no issues at all.

**Comment by ThatSlimGuy** [comment]
This is something super strange as well, i went out of my way to install kodi onto a new tablet, and put one episode of family guy on there, installed trakt addon, and linked my account. Played the show and with about 20 seconds to go, stopped the show, and yet again, no rating popup comes up, marks show as watched in Kodi, nothing on trakt, but it kept the file as still on 97% through on trakt as live scrobble.... Thats super odd right @rectifyer ?

**Comment by onlytanmoy** [comment]
The Trakt rating pop-up that comes after watching the movie/show no more comes up now :(
I am using Shield TV Pro 2019 with Arctic Horizon Kodi skin. I reckon the Trakt program add-on is not working correctly. The scrobbling is handled via the video add-on being used to watch the movie/show say Seren, Shadow...so after I am done watching 85% of the content, its getting registered in Trakt history, no problem there. However the rating pop-up (handled via the Trakt addon) is not showing up at the end. I also tried one more addon developed by Razzeee, its called Context menus - Trakt- Contextmenu; also not working. I know most people find it silly to rate a movie/show, however I enjoy it, so will be great to have this fixed.

**Comment by razzeee** [comment]
I just tested Leia on windows and got a ratings dialog for both movie and
show. While stopping each one before it reached the end.

**Comment by ThatSlimGuy** [comment]
Can you test Android ?

**Comment by onlytanmoy** [comment]
@Razzeee I am not using Kodi on Windows but on Android device Shield. Can you pls check the behaviour on Android?

**Comment by razzeee** [comment]
I'm already on matrix on my shield and watching till the end works fine, will try to stop prematurely

**Comment by onlytanmoy** [comment]
> I'm already on matrix on my shield and watching till the end works fine, will try to stop prematurely

I am using Kodi 18.7 Leia on my shield 🙂

**Comment by razzeee** [comment]
Matrix seems fine to me in that regard. There are some other things that I need to look at for matrix.

**Comment by ThatSlimGuy** [comment]
You saw the logs so I'm not exactly sure what the issue could be. I've tried literally everything I can think of to be able to get around the issue but it always happens

**Comment by razzeee** [comment]
Your also not using any DNS right?

**Comment by ThatSlimGuy** [comment]
100% not using DNS at all. Never had any issue viewing website on the same network on a PC or anything.

**Comment by onlytanmoy** [comment]
> Your also not using any DNS right?

Correct, not using any DNS or VPN.

**Comment by ThatSlimGuy** [comment]
> Your also not using any DNS right?

Can confirm still an issue. Was watching mainly tv shows but watched a movie today. Came to the end with 3 minutes to go, says on trakt website via scrobble the show is 96% in, I have the setting set to confirm rating after 90% press stop on show, sets movie as watched in Kodi, on trakt website it stays on the scrobble as if I'm still watching the movie, never comes up with rating, never allows the adding as watched to trakt at all.

Is there any chance that having the smb settings set to smbv3 or possibly having ipv6 enabled on my modem that could be causing these issues ?

**Comment by razzeee** [comment]
well, if we're comparing setups yes. but it shouldn't matter in reality, as smb isn't involved at all in scrobbling and ipv6 is handled by the requests lib and I would be very surprised if we found an error there.

**Comment by ThatSlimGuy** [comment]
> well, if we're comparing setups yes. but it shouldn't matter in reality, as smb isn't involved at all in scrobbling and ipv6 is handled by the requests lib and I would be very surprised if we found an error there.

You're also on windows as well right now Android ?

**Comment by razzeee** [comment]
My mainsetup is android tv.

**Comment by onlytanmoy** [comment]
> My mainsetup is android tv.

And the rating pop up shows fine there?
I know it worked for me in 18.6, something probably broke in 18.7 Leia 😐

**Comment by ThatSlimGuy** [comment]
> > My mainsetup is android tv.
> 
> And the rating pop up shows fine there?
> I know it worked for me in 18.6, something probably broke in 18.7 Leia

The strange thing is though you have your items marking are watched on trakt still tho right ? Is that only after a library refresh? Like when you finish watching something, does anything popup saying it's now been marked as watched on trakt ?

**Comment by onlytanmoy** [comment]
No, I don't get any pop up saying item marked as watched in Trakt. However all my items do get registered in Trakt as watched. The only issue I am facing is the missing rating pop up dialog.

**Comment by ThatSlimGuy** [comment]
Have you ever checked trakt as soon as you finishing watching something to see if it shows up instantly ?

**Comment by onlytanmoy** [comment]
> Have you ever checked trakt as soon as you finishing watching something to see if it shows up instantly ?

Yep, it shows up instantly in my Trakt history. I mostly use Seren to play movies/shows. What about you?

**Comment by razzeee** [comment]
Yes, rating popup shows fine for me. But I'm only using local content. So that might in fact be your problem.

**Comment by onlytanmoy** [comment]
> Yes, rating popup shows fine for me. But I'm only using local content. So that might in fact be your problem.

Hmm..I see..can you please try playing some online content via any video add-on in Kodi and see if the rating pop up appears then?

**Comment by ThatSlimGuy** [comment]
Mmmm I only use local content as well. Extremely frustrating. Watched 7-8 episodes today from various shows, also 2 movies, no ratings popup ever. I can only have my items show up as watched on twitch when doing a refresh of library which then does a trakt sync after

**Comment by onlytanmoy** [comment]
Oh ok, me on the other hand not using any local content 🙂

**Comment by ThatSlimGuy** [comment]
Guessing not much movement on this issue ?

**Comment by onlytanmoy** [comment]
@ThatSlimGuy @Razzeee The Trakt rating pop up started appearing mysteriously now but only for TV episodes, which is great ☺️ Still no luck for movies though ☹️ I checked in Mibox and Shield, same behaviour witnessed in both the devices.

**Comment by ThatSlimGuy** [comment]
> @ThatSlimGuy @Razzeee The Trakt rating pop up started appearing mysteriously now but only for TV episodes, which is great ☺️ Still no luck for movies though ☹️ I checked in Mibox and Shield, same behaviour witnessed in both the devices.

Ive been using my setup with the older versions and such installed ever since, trakt still never ever marks anything as watched, can only get it to sync by running it manually or doing a library update. Is annoying, but is only way to get it to work. Doing it with updated version will either not allow movies to be synced or added, or just won't work at all. So surprised no one else is facing these issues. Don't know where it all went bad.

**Comment by onlytanmoy** [comment]
I tested in Kodi 18.6. Pop-up came for TV episode but absent for movies. So Kodi 18.7 is not the problem.
@Razzeee is there any way I can downgrade the current Trakt version of 3.2.5?

**Comment by razzeee** [comment]
@onlytanmoy you should be able to downgrade from the addon screen, when you hit the update button. But I think you need to have had an old version installed at one point for that to work. Otherwise you need to download a version from here for example.

**Comment by ThatSlimGuy** [comment]
Downgrading was the absolute only way I could get the addon to work in any way. Looks like it's going to be the only way to have it work from here on out

**Comment by onlytanmoy** [comment]
@ThatSlimGuy I tried with version 3.2.4, same behaviour..rating pop-up coming for TV episodes alone! Which version worked for you? Please share download link.

**Comment by onlytanmoy** [comment]
@Razzeee how to download previous versions from here? Please guide.

**Comment by ThatSlimGuy** [comment]
I think I have to downgrade 2 different things to get mine to detect movies again despite it not adding movies to my watched list. I think it was the main script plus one of the files it relies on. When I can I'll post what versions I had to use for it to work

**Comment by razzeee** [comment]
Likely modules.trakt.py

**Comment by onlytanmoy** [comment]
@ThatSlimGuy can you go to your Kodi My add-ons, program add-on and check the Trakt version number for me (the one that is working)?

**Comment by ThatSlimGuy** [comment]
> @ThatSlimGuy can you go to your Kodi My add-ons, program add-on and check the Trakt version number for me (the one that is working)?

So sorry for the delay, been super busy over last few weeks.

Trakt 3.2.0 
Trakt.py

those 2 things are what i had to change to allow even movies to be found when trying to mark anything off. I noticed that even when running the addon with any of the new versions, it would get to 17% into movies and then it would just not go any further. 

Those to versions above while still to this day refuse to mark anything as watched unless u watch the movie or show right until the very very end, it still allows it to semi work.

**Comment by ThatSlimGuy** [comment]
I had to downgrade the trakt.py dependency as well as even downgrading trakt to lower version even as a brand new fresh trakt install, it still wouldnt allow movies to be mark in library.

**Comment by onlytanmoy** [comment]
@ThatSlimGuy Thanks for getting back to me. Since the only issue at my end is the Trakt rating dialog not coming after watching a movie, I will stick to my current setup. I am using a Trakt app in my smart devices to rate the movie as a work around.

**Comment by ThatSlimGuy** [comment]
> @ThatSlimGuy Thanks for getting back to me. Since the only issue at my end is the Trakt rating dialog not coming after watching a movie, I will stick to my current setup. I am using a Trakt app in my smart devices to rate the movie as a work around.

I thought you were having another issue for it marking the movie as watched within Kodi as well?

**Comment by onlytanmoy** [comment]
Nope, just the rating pop-up not coming after done watching a movie.
So, both movies and episodes getting marked as watched & rating pop-up showing up for episodes alone.
